### PR TITLE
Fix StorageProfile ClaimPropertySets validation

### PR DESCRIPTION
**Pull request by arnongilboa**
**What this PR does / why we need it**:
We have no reason to allow setting `StorageProfile` `ClaimPropertySet` missing either `VolumeMode` or `AccessModes`.

**Which issue(s) this PR fixes**:
Jira-ticket: https://issues.redhat.com/browse/CNV-37991

**Release note**:
```release-note
BugFix: StorageProfile ClaimPropertySets validation
```

### CODE REVIEW

File: pkg/controller/storageprofile-controller.go [hunk 1]
```diff
@@ -101,8 +102,8 @@ func (r *StorageProfileReconciler) reconcileStorageProfile(sc *storagev1.Storage
 
 	if len(storageProfile.Spec.ClaimPropertySets) > 0 {
 		for _, cps := range storageProfile.Spec.ClaimPropertySets {
-			if len(cps.AccessModes) == 0 && cps.VolumeMode != nil {
-				err = fmt.Errorf("must provide access mode for volume mode: %s", *cps.VolumeMode)
+			if cps.VolumeMode == nil || len(cps.AccessModes) == 0 {
```
Comment (mhenriks) [inline pkg/controller/storageprofile-controller.go:105]: `cps.VolumeMode == nil` is redundant.  If it is nil len(cps.AccessModes) will return 0

Comment (arnongilboa) [reply pkg/controller/storageprofile-controller.go:105]: How is `cps.VolumeMode == nil` related to  `len(cps.AccessModes)`?

File: staging/src/kubevirt.io/containerized-data-importer-api/pkg/apis/core/v1beta1/types.go [hunk 2]
```diff
@@ -445,12 +445,11 @@ type StorageProfileStatus struct {
 type ClaimPropertySet struct {
 	// AccessModes contains the desired access modes the volume should have.
 	// More info: https://kubernetes.io/docs/concepts/storage/persistent-volumes#access-modes-1
-	// +optional
-	AccessModes []corev1.PersistentVolumeAccessMode `json:"accessModes,omitempty" protobuf:"bytes,1,rep,name=accessModes,casttype=PersistentVolumeAccessMode"`
+	AccessModes []corev1.PersistentVolumeAccessMode `json:"accessModes"`
```
Comment (arnongilboa) [inline staging/src/kubevirt.io/containerized-data-importer-api/pkg/apis/core/v1beta1/types.go:448]: afaik there's no `kubebuilder:validation` marker for validating enum values for a slice (`AccessModes`) like addded for `VolumeMode`. The only ways I can think of is either by a validating webhook, or by adding an alias type, e.g.:
```go
+// +kubebuilder:validation:Enum="ReadWriteOnce";"ReadOnlyMany";"ReadWriteMany";"ReadWriteOncePod"
+type ClaimPropertySetAccessMode corev1.PersistentVolumeAccessMode
```
Both solutions aren't so nice.

Comment (awels) [reply staging/src/kubevirt.io/containerized-data-importer-api/pkg/apis/core/v1beta1/types.go:448]: I don't mind the second option, creating a type that is just the same, and adding the validation there.

**Review by awels — APPROVED** [review]

File: pkg/controller/storageprofile-controller.go (hunk 1 above)
Comment (mhenriks) [reply pkg/controller/storageprofile-controller.go:105]: lol it's not

**Pull request merged**